src/common/btree.c

```diff
@@ -901,8 +913,14 @@ btr_root_grow(struct btr_context *tcx, umem_off_t off_left,
 	at = !btr_node_is_equal(tcx, off_left, tcx->tc_trace->tr_node);
 
 	/* replace the root offset, increase tree level */
-	if (btr_has_tx(tcx))
-		btr_root_tx_add(tcx); /* XXX check error */
+	if (btr_has_tx(tcx)) {
+		rc = btr_root_tx_add(tcx); /* XXX check error */
+		if (rc != 0) {
+			D_ERROR("Failed to add root into TX: %s\n",
+				strerror(errno));
+			return rc;
+		}
```

Comment:
Same as my comment on btr_root_start.

Reply:
Same comment.  Perhaps we could assert rc == 0 || btr_umm(tcx)->umm_id != UMEM_CLASS_VMEM